src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -92,6 +94,39 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
                     replacedText[0] = '[';
                     replacedText[replacedText.Length - 1] = ']';
                     bracketLiteralText = replacedText.ToString();
+
+                    Debug.Assert(expectedType.IsCollection());
+                    if (expectedType.Definition.AsElementType().FullTypeName().Equals("Edm.String"))
+                    {
+                        // For collection of strings, need to convert single-quoted string to double-quoted string,
+                        // and also, per ABNF, two consecutive single quotes  to one single quote.
+                        // Sample: ['a''bc','''def','xyz'''] ==> ["a'bc","'def","xyz'"], which is legitimate Json format.
+                        string[] items = bracketLiteralText.Substring(1, bracketLiteralText.Length - 2).Split(',');

```

Comment:
Just a heads-up, when hardcoded complex type collections are allowed in the future, I think this line will have to be changed.

Reply:
I think I know what you are referring to. I heard Mike mentioned that IN operator collection will be changed to using brackets instead of parenthesis. Thanks for the heads up.